### ISSUE TITLE
federation e2e tests failing consistently with gce/util.sh: line 1192: local: `replica-pd=jenkins-us-central1-b-master-pd': not a valid identifier

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#32997]
./hack/e2e-internal/.../cluster/gce/util.sh: line 1192: local: `replica-pd=jenkins-us-central1-b-master-pd': not a valid identifier

@colhom @madhusudancs FYI - this looks related to your recent work?  I don't see any obvious PR to blame, but I seem to remember you guys touching this part of the code recently?

BTW, I deleted a bunch of leaked resources in the relevant test project on Friday (9/16), assuming that the test scripts would re-create whatever they needed.  It's possible (but unlikely) that this has anything to do with that.

**Comment by quinton-hoole** [comment]
@colhom Here are some example logs to aid debugging:

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gce-federation/4332

**Comment by colhom** [comment]
@quinton-hoole I believe this to be the source of dysfunction:

```sh
ERROR: (gcloud.compute.disks.create) Some requests did not succeed:
 - The resource 'projects/k8s-jkns-e2e-gce-federation/zones/us-central1-a/disks/jenkins-us-central1-a-master-pd' already exists
```

Which is turn caused by what you found, which causes the disk to not be cleaned up ahead of time: 

```sh
./hack/e2e-internal/.../cluster/gce/util.sh: line 1192: local: `replica-pd=jenkins-us-central1-b-master-pd': not a valid identifier
```
https://github.com/kubernetes/kubernetes/blob/87b40782c9c1cc82ba42f75de4cf23126297e0a3/cluster/gce/util.sh#L1194

You can try this out in your local shell:
```sh
[chom@hompad-work kubernetes]$ REPLICA_NAME="replica-pd=jenkins-us-central1-a-master"
[chom@hompad-work kubernetes]$ function set-replica-name { local -r replica-pd="${REPLICA_NAME:-${MASTER_NAME}}-pd"; }
[chom@hompad-work kubernetes]$ set-replica-name 
-bash: local: `replica-pd=replica-pd=jenkins-us-central1-a-master-pd': not a valid identifier
```

git blames leads us to this recent commit: https://github.com/kubernetes/kubernetes/commit/529818c9

\cc the author @jszczepkowski

The root problem is unfortunatley very annoying: this is how bash tells you that **variable names (in this case, $replica-pd) cannot contain dashes**!

**Comment by colhom** [comment]
@quinton-hoole fix is up in #33039

**Comment by quinton-hoole** [comment]
Awesome @colhom many thanks.  I'll make sure it gets merged ASAP.
